java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -617,6 +633,22 @@ private static <T> Optional<T> or(Optional<T> opt, Supplier<? extends Optional<?
         }
     }
 
+    private static ByteArrayInputStream toByteArrayInputStream(InputStream is) throws IOException {
+        // Optimization to avoid copying when applicable. `executeAsync` will produce `ByteArrayInputStream`.
+        if (is instanceof ByteArrayInputStream) {
+            return (ByteArrayInputStream) is;
+        }
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+        int len;
+        while ((len = is.read(buffer)) > -1) {
+            baos.write(buffer, 0, len);
+        }
+        baos.flush();
+        return new ByteArrayInputStream(baos.toByteArray());
```

Comment:
We could re-use aws-sdk-java's `IoUtils` to shorten this: https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/IoUtils.java#L39-L52
```suggestion
        return new ByteArrayInputStream(IoUtils.toByteArray(is));
```

Reply:
Oh awesome. This definitely removes verbosity 💯 .